crates/astria-sequencer-types/src/sequencer_block_data.rs

```diff
@@ -43,9 +43,17 @@ pub enum Error {
 pub struct ChainId(#[serde(with = "hex::serde")] Vec<u8>);
 
 impl ChainId {
-    #[must_use]
-    pub fn new(inner: Vec<u8>) -> Self {
-        Self(inner)
+    /// Creates a new `ChainId` from the given bytes.
+    ///
+    /// # Errors
+    ///
+    /// - if the given bytes are longer than 32 bytes
+    pub fn new(inner: Vec<u8>) -> eyre::Result<Self> {
```

Comment:
In PR #376 eyre is removed in favor of typing the errors. Can you do that here as well and merge after that PR?

Reply:
yep, will wait until that's merged!